src/checkmatelib/client.py

```diff
@@ -88,12 +88,17 @@ def check_url(  # pylint: disable=inconsistent-return-statements
     def _clean_url(cls, url):
         """Clean the URL before we send it.
 
-        Checkmate will do this for itself, and doesn't trust our input, but
-        this allows us to fail fast for bad URLs, before Checkmate has a chance
-        to check them. We can also apply extra constraints.
+        This applies the same canonicalization to the URL as we do when the
+        real check happens in Checkmate. This means anything which will cause
+        Checkmate to raise `BadURL` will cause us to first, saving some time.
+
+        We also apply a check that the domain must be public, baring some
+        domains used for local testing. There's no point in going to Checkmate
+        for these. We can use them.
```

Comment:
```suggestion
        for these. We can use them.
```


Do you mean we _cannot_ use this domains / we already know they will fail?

Reply:
Yeah typo